app/src/test/java/org/mozilla/fenix/components/PrivateShortcutCreateManagerTest.kt

```diff
@@ -0,0 +1,83 @@
+
+@RunWith(FenixRobolectricTestRunner::class)
+class PrivateShortcutCreateManagerTest {
+
+    @Test
+    fun `GIVEN shortcut pinning is not supported WHEN createPrivateShortcut is called THEN do not create a pinned shortcut`() {
+        val privateShortcutCreateManager = spyk<PrivateShortcutCreateManager>()
+        mockkStatic(ShortcutManagerCompat::class)
+
+        every { ShortcutManagerCompat.isRequestPinShortcutSupported(testContext) } returns false
+
+        privateShortcutCreateManager.createPrivateShortcut(testContext)
+
+        verify(exactly = 0) { ShortcutManagerCompat.requestPinShortcut(testContext, any(), any()) }
+    }
+
+    @Test
+    fun `GIVEN shortcut pinning is supported WHEN createPrivateShortcut is called THEN create a pinned shortcut`() {
+        val shortcut = slot<ShortcutInfoCompat>()
+        val intentSender = slot<IntentSender>()
+        val intent = slot<Intent>()
+
+        val privateShortcutCreateManager = spyk<PrivateShortcutCreateManager>()
+        mockkStatic(ShortcutManagerCompat::class)
+        mockkStatic(PendingIntent::class)
+
+        every { ShortcutManagerCompat.isRequestPinShortcutSupported(testContext) } returns true
+
+        privateShortcutCreateManager.createPrivateShortcut(testContext)
+
+        verify { PendingIntent.getActivity(testContext, 0, capture(intent), PendingIntent.FLAG_UPDATE_CURRENT) }
+        verify { ShortcutManagerCompat.requestPinShortcut(testContext, capture(shortcut), capture(intentSender)) }
+        `assert shortcutInfoCompat is build correctly`(shortcut, testContext)
+        `assert homeScreenIntent is built correctly`(intent)
+    }
+
+    private fun `assert shortcutInfoCompat is build correctly`(shortcutInfoCompat: CapturingSlot<ShortcutInfoCompat>, context: Context) {
+        shortcutInfoCompat.apply {
+            assertEquals(context.getString(R.string.app_name_private_5, context.getString(R.string.app_name)), captured.shortLabel)
+            assertEquals(context.getString(R.string.app_name_private_5, context.getString(R.string.app_name)), captured.longLabel)
+            assertEquals(R.mipmap.ic_launcher_private_round, captured.icon.resId)
+            `assert homeActivity intent is built correctly`(captured.intent)
+        }
+    }
+
+    private fun `assert homeActivity intent is built correctly`(intent: Intent) {
+        assertEquals(Intent.ACTION_VIEW, intent.action)
+        assertEquals(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK, intent.flags)
+        assertEquals(HomeActivity::class.qualifiedName, intent.component?.className)
+        assertEquals(true, intent.extras?.getBoolean(HomeActivity.PRIVATE_BROWSING_MODE))
+        assertEquals(StartSearchIntentProcessor.PRIVATE_BROWSING_PINNED_SHORTCUT, intent.extras?.getString(HomeActivity.OPEN_TO_SEARCH))
+    }
+
+    private fun `assert homeScreenIntent is built correctly`(intent: CapturingSlot<Intent>) {
+        intent.apply {
```

Comment:
The same as above. Maybe `with` is a more appropriate scoping function to use here or maybe we don't even need one.

Reply:
Yes, after sending captured value directly. It does not make sense 
I have updated the code